### PR TITLE
OpenCL 2.1 Package

### PR DESCRIPTION
**Pull request by caseymcc**
Add Opencl 2.1 package, includes headers and ICD. Current package hosted in my repo, but will post a PR to hunter_packages/opencl when available.

Testing repo at: https://github.com/caseymcc/hunter/tree/pkg.opencl
Changes from template:
-Apple not supported (including iOs) (only OpenCl 1.2 seems to be supported by Apple)
-Android not supported
-Visual Studio 9-11 not supported (headers include const expressions)
-mingw (on Windows) not supported (links against native directX which has some Microsoft specific code that gcc does not understand).

Latest Test:
https://travis-ci.org/caseymcc/hunter/builds/176831259
https://ci.appveyor.com/project/caseymcc/hunter/build/1.0.29

Considering nVidia does not support OpenCL 2.1 I will go back a little later and add a 1.2 Version (hopefully get it working with Apple then)

### CODE REVIEW

File: cmake/projects/OpenCL/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# !!! DO NOT PLACE HEADER GUARDS HERE !!!
+
+# Load used modules
+include(hunter_add_package)
+include(hunter_add_version)
+include(hunter_cacheable)
+include(hunter_cmake_args)
```
Comment (ruslo) [inline cmake/projects/OpenCL/hunter.cmake]: Not used.

File: cmake/projects/OpenCL/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# List of versions here...
+
+hunter_add_version(
+    PACKAGE_NAME
+    OpenCL
+    VERSION
+    "2.1-p0"
+    URL
+    "https://github.com/caseymcc/OpenCL-ICD-Loader/archive/2.1-p0.tar.gz"
+#    "D:/projects/hunter_cmake/packages/OpenCL-ICD-Loader"
```
Comment (ruslo) [inline cmake/projects/OpenCL/hunter.cmake]: To remove.

File: cmake/projects/OpenCL/hunter.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    SHA1
+    bb460fabd67a7550798b413244528fd8177dfc6c
+)
+
+#hunter_cmake_args(
```
Comment (ruslo) [inline cmake/projects/OpenCL/hunter.cmake]: To remove

File: cmake/projects/OpenCL/hunter.cmake [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#    OpenCL
+#    CMAKE_ARGS
+#)
+
+# Pick a download scheme
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+hunter_cacheable(OpenCL)
+#hunter_pick_scheme(DEFAULT url_local_cmake)
```
Comment (ruslo) [inline cmake/projects/OpenCL/hunter.cmake]: To remove

**Comment by ruslo** [comment]
> but will post a PR to hunter_packages/opencl when available

Created https://github.com/caseymcc/OpenCL-ICD-Loader . You're invited (check your email). Read this: https://docs.hunter.sh/en/latest/creating-new/cmake-dependencies.html#maintenance

**Comment by ruslo** [comment]
@headupinclouds +cc

**Comment by ruslo** [comment]
> -Apple not supported (including iOs) (only OpenCl 1.2 seems to be supported by Apple)
> -Android not supported
> -Visual Studio 9-11 not supported (headers include const expressions)
> -mingw (on Windows) not supported (links against native directX which has some Microsoft specific code that gcc does not understand).

I will create pkg.opencl later and you can send the pull request with excluded toolchains and descriptions. I will start with the full set of toolchains.

**Comment by headupinclouds** [comment]
> -Android not supported

@caseymcc : Is there a fundamental blocking issue for Android, or is this just not needed?

**Comment by caseymcc** [comment]
It is not needed for Android, for Android you have to link against the NDK which should have both the header files and the lib.

**Comment by ruslo** [comment]
@caseymcc ready to be tested?

**Comment by caseymcc** [comment]
Should be good to go.

**Comment by ruslo** [comment]
> Should be good to go

Great, testing now:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1206
* https://travis-ci.org/ingenue/hunter/builds/180306949

**Comment by ruslo** [comment]
@caseymcc Some toolchains failed as expected (matches your list). Please send pull request to [pkg.opencl](https://github.com/ingenue/hunter/tree/pkg.opencl) branch which will exclude broken toolchains. Please add any useful info about status in comments.

**Comment by ruslo** [comment]
New testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1228
* https://travis-ci.org/ingenue/hunter/builds/181623064

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1229
* https://travis-ci.org/ingenue/hunter/builds/181653764

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.17.3

@caseymcc feel free to add new wiki page

**Pull request merged**